gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/LifecycleTest.kt

```diff
@@ -0,0 +1,90 @@
+package com.squareup.anvil.plugin
+
+import com.rickbusarow.kase.gradle.dsl.buildFile
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldInclude
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.TestFactory
+
+class LifecycleTest : BaseGradleTest() {
+
+  @TestFactory
+  fun `tasks are compatible with configuration caching when targeting kotlin-jvm`() =
+    kases.distinctBy { it.gradleVersion }
+      .asTests {
+
+        rootProject {
+          buildFile {
+            plugins {
+              kotlin("jvm", it.kotlinVersion)
+              id("com.squareup.anvil", anvilVersion)
+            }
+            dependencies {
+              compileOnly(libs.inject)
+            }
+          }
+
+          dir("src/main/java") {
+            injectClass()
+          }
+        }
+
+        val expected = when {
+          it.gradleVersion >= "8.5" -> "Calculating task graph as no cached configuration is available for tasks: compileKotlin"
+          it.gradleVersion >= "8.4" -> "Calculating task graph as no configuration cache is available for tasks: compileKotlin"
+          else -> "0 problems were found storing the configuration cache."
+        }
+
+        shouldSucceed(
+          "compileKotlin",
+          "--configuration-cache",
+          withHermeticTestKit = true,
+        ) {
+          output shouldInclude expected
+        }
+        shouldSucceed(
+          "compileKotlin",
+          "--configuration-cache",
+          withHermeticTestKit = true,
+        ) {
+          output shouldInclude "Reusing configuration cache."
+        }
+      }
+
+  @TestFactory
+  fun `compileKotlin is up to date when no changes are made`() = testFactory {
+
+    rootProject {
+      dir("src/main/java") {
+        injectClass()
+      }
+    }
+
+    shouldSucceed("compileKotlin")
+    shouldSucceed("compileKotlin") {
+      task(":compileKotlin")?.outcome shouldBe TaskOutcome.UP_TO_DATE
+    }
+  }
+
+  @TestFactory
+  fun `compileKotlin is FROM-CACHE after the project build directory is deleted`() = testFactory {
```

Comment:
I love how clean and simple these test cases are

Reply:
Thanks!  That's the idea. :)